### ISSUE TITLE
Add URL routing

### ISSUE THREAD

**Issue by davidgatti** [0x4447-office/0x4447_webpage_documentation#58]
I updated the CloudFront settings to redirect all the URL back to the index.html file. So when you go to a product page, and then refresh you will see the home page. Instead when you refresh the page it should stay on the URL that you are on.

**Comment by acidbiscuit** [comment]
@davidgatti It sounds like configuration you described is for single page applications (SPA), which has single entry HTML file and client-side routing is available. Such configuration will not work with server side rendering (SSR), which is used by Vuepress to keep all benefits of SEO.

What problem are you trying to solve with such configuration?

**Comment by davidgatti** [comment]
There is no server for this project, I'm building the site and putting it on S3, you can see what I do here: https://github.com/0x4447-office/0x4447_webpage_documentation/blob/master/buildspec.yml#L11. 

The problem right now is this:

1. open a private windo
2. paste in the URL bar this: https://documentation.0x4447.com/samba/1.2.0/
3. You will see the home page, and not the product page.

I'd like to see the product page not the home page. 

Since we have the AVC page done in nuxt and is SEO friendly because each page is generated, can we have the same here?

**Comment by acidbiscuit** [comment]
@davidgatti If all build files are present on s3, then it is likely that something is off in CloudFront configuration. Vuepress build outputs a static `index.html` file for every page and no additional configuration needed. All the rules of serving those files should be configured on the web server. Client-side routing picks up after page has been rendered and works correctly on https://documentation.0x4447.com/samba/1.2.0/. It looks like initial request to /samba/1.2.0/index.html is being redirected to the root index.html. Can you, please, double check?

**Comment by davidgatti** [comment]
OK, so after some more testing, this is what is happening at least on my side:

- Visting https://documentation.0x4447.com/samba/1.2.0/
- get the home page
- refresh once, get the product page. 

As far as I know CloudFront is configured in a way that it looks for an index file in a folder, and if it finds it disply it.

I will double check, but can you check if what is happening is related with the framework itself, where maybe the routing is loaded not in the right moment?

**Comment by acidbiscuit** [comment]
@davidgatti I've tested on a simple server:
- The output of the app build creates a valid folder structure for every page with corresponding HTML file
- Navigation on client-side (after you click a router link on the page) is working correctly. 

My observation from https://documentation.0x4447.com/samba/1.2.0/
- Web server handles 404 requests correctly
- Any valid page URL serves root `/index.html` (homepage) instead of `index.html` with request path, e.g. `/samba/1.2.0/index.html`.

**Comment by davidgatti** [comment]
> * The output of the app build creates a valid folder structure for every page with corresponding HTML file
> * Navigation on client-side (after you click a router link on the page) is working correctly.

Yes, this works, but the issues shows up when you open a new browser with not cashed content, copy and past a direct URL, clicking around works.

> * Any valid page URL serves root `/index.html` (homepage) instead of `index.html` with request path, e.g. `/samba/1.2.0/index.html`.

Will look in to this.
